napari/components/_viewer_key_bindings.py

```diff
@@ -59,15 +58,22 @@ def toggle_ndisplay(viewer: Viewer):
 @register_viewer_action(trans._("Toggle theme."))
 def toggle_theme(viewer: Viewer):
     """Toggle theme for viewer"""
-    settings = get_settings()
     themes = available_themes()
-    current_theme = settings.appearance.theme
+    current_theme = viewer.theme
+    # Check what the system theme is, to toggle properly
+    if current_theme == 'system':
+        current_theme = get_system_theme()
     idx = themes.index(current_theme)
     idx += 1
     if idx == len(themes):
         idx = 0
```

Comment:
```suggestion
    idx = (idx + 1) % len(themes)
```

I'm not sure if it is more pythonic, but it is shorter. It is a simple modulo calculation.

Reply:
That's pretty baller, I had to test it a buncha times to believe it.
I wonder if that hinders readability? At least to me, the modulo, while a concept I understand, doesn't immediately make clear what's  going on.
🤷‍♂️

Reply:
The main how it increases readability is that it increases the visible part of code without scrolling.

Reply:
But this method does prevent `idx` from ever being `>len(themes)`, so it does eliminate the nested `if`, if I'm thinking about it correctly.

Reply:
err.. wait, no because if idx is len-1, the last element, and that is `system`, then when we skip system with `idx += 1`, idx > len. So the nested `if` is still needed.

Reply:
No. An increase for the "system" could be replaced by a second modulo operation. I also add a second proposition.

Reply:
ok ok, I get it. Sorry, I'm slow.